### PR TITLE
Tab to switch toolbar slots

### PR DESCRIPTION
**Pull request by masoo12**
Adds inventory sounds, for drag and drop and allow users to press tab to display the 10 items in the inventory in the toolbar. #656

### CODE REVIEW

File: source/core/src/main/com/csse3200/game/components/inventory/ToolbarDisplay.java [hunk 1]
```diff
@@ -243,4 +254,14 @@ private void updateItemSlot(int slotNum) {
...
+    private void switchToolbar() {
+        window.addAction(sequence(scaleTo(1.02f, 1.02f, 0.2f, Interpolation.fade),scaleTo(1, 1, 0.2f, Interpolation.fade)));
+        try {
+            ServiceLocator.getSoundService().getEffectsMusicService().play(EffectSoundFile.SWITCH_TOOLBAR);
+        } catch (InvalidSoundFileException e) {
+            logger.error("sound not loaded");
+        }
+        curTab = (curTab + 1) % 3;
+        updateToolbar();
+    }
```
Comment (tristanduncombe) [inline source/core/src/main/com/csse3200/game/components/inventory/ToolbarDisplay.java:266]: What does this do?

This seems just to make a noise but no have any function. It seems as though it should cycle through the rows, but it doesn't do that

File: source/core/src/main/com/csse3200/game/components/items/ItemComponent.java [hunk 2]
```diff
@@ -30,18 +30,6 @@ public ItemComponent(String itemName, ItemType itemType, String texturePath) {
...
-	 *
-	 * @param itemName user facing name for item
-	 * @param itemType the enum for type of item
-	 * @param perishable whether the item is perishable
-	 * @param texturePath path for the item texture
-	 */
-	public ItemComponent(String itemName, ItemType itemType, String texturePath, boolean perishable) {
-		this(itemName, itemType, "", 0, false, perishable, texturePath);
-	}
-
```
Comment (tristanduncombe) [inline source/core/src/main/com/csse3200/game/components/items/ItemComponent.java:44]: Why is this removed?

File: source/core/src/main/com/csse3200/game/components/items/ItemComponent.java [hunk 3]
```diff
@@ -66,18 +54,6 @@ public ItemComponent(String itemName, ItemType itemType, String itemDescription,
...
-	 *
-	 * @param itemName        user facing name for item
-	 * @param itemType        the enum for type of item
-	 * @param itemDescription user facing description for item
-	 * @param perishable whether the item is perishable
-	 * @param texturePath path for the item texture
-	 */
-	public ItemComponent(String itemName, ItemType itemType, String itemDescription, boolean perishable, String texturePath) {
-		this(itemName, itemType, itemDescription, 0, false, perishable, texturePath);
-	}
```
Comment (tristanduncombe) [inline source/core/src/main/com/csse3200/game/components/items/ItemComponent.java:80]: Why is this removed?

File: source/core/src/test/com/csse3200/game/components/inventory/TestToolbarUI.java [hunk 4]
```diff
@@ -164,16 +163,16 @@ private static Stream<Arguments> addingItemsShouldAddInventoryImagesParams() {
...
-				arguments(new ItemComponent("Hoe", ItemType.HOE, "images/tool_hoe.png"), 0),
-				arguments(new ItemComponent("Scythe", ItemType.SCYTHE, "images/tool_scythe.png"), 1),
-				arguments(new ItemComponent("Shovel", ItemType.SHOVEL, "images/tool_shovel.png"), 2),
-				arguments(new ItemComponent("Item", ItemType.FERTILISER, "images/tool_shovel.png"), 3),
-				arguments(new ItemComponent("Hoe", ItemType.HOE, "images/tool_hoe.png"), 4),
-				arguments(new ItemComponent("Scythe", ItemType.SCYTHE, "images/tool_scythe.png"), 5),
-				arguments(new ItemComponent("Shovel", ItemType.SHOVEL, "images/tool_shovel.png"), 6),
-				arguments(new ItemComponent("Item", ItemType.FERTILISER, "images/tool_shovel.png"), 7),
-				arguments(new ItemComponent("Hoe", ItemType.HOE, "images/tool_hoe.png"), 8),
-				arguments(new ItemComponent("Scythe", ItemType.SCYTHE, "images/tool_scythe.png"), 9)
```
Comment (tristanduncombe) [inline source/core/src/test/com/csse3200/game/components/inventory/TestToolbarUI.java:176]: Why is this modified?

File: source/core/src/main/com/csse3200/game/components/inventory/InventoryDisplay.java [hunk 5]
```diff
@@ -215,6 +220,11 @@ public DragAndDrop.Payload dragStart(InputEvent event, float x, float y, int poi
 				@Override
 				public void dragStop(InputEvent event, float x, float y, int pointer, DragAndDrop.Payload payload, DragAndDrop.Target target) {
 					if (target == null) {
+						try {
+							ServiceLocator.getSoundService().getEffectsMusicService().play(EffectSoundFile.DROP_ITEM);
+						} catch (InvalidSoundFileException e) {
+							logger.error("sound not loaded");
```
Comment (tristanduncombe) [inline source/core/src/main/com/csse3200/game/components/inventory/InventoryDisplay.java:226]: Is is possible to only play it if it successfully moved or swaps position?

**Review by tristanduncombe — CHANGES_REQUESTED** [review]
Needs work - confusing as to what you are trying to achieve.

File: source/core/src/main/com/csse3200/game/components/items/ItemComponent.java (hunk 2 above)
Comment (masoo12) [reply source/core/src/main/com/csse3200/game/components/items/ItemComponent.java:44]: I removed all references to perishable items because its not being used .

File: source/core/src/main/com/csse3200/game/components/inventory/ToolbarDisplay.java (hunk 1 above)
Comment (tristanduncombe) [reply source/core/src/main/com/csse3200/game/components/inventory/ToolbarDisplay.java:266]: Good job on the improvements, but rather than just changing what the toolbar displays, could you rather update the inventory component. This will mean the change is updated in the inventoryDisplay which will help the user understand the change.

**Review by tristanduncombe — CHANGES_REQUESTED** [review]
Let me know if this is feasible.

File: source/core/src/main/com/csse3200/game/components/player/InventoryComponent.java [hunk 6]
```diff
@@ -197,6 +198,9 @@ public Boolean hasItem(String item) {
     }
 
     private void newInventory() {
+        if (entity != null) {
+            entity.getEvents().addListener("switchToolbar",this::switchTab);
```
Comment (nimeshgarg) [inline source/core/src/main/com/csse3200/game/components/player/InventoryComponent.java:202]: Why are you adding a listener inside newInventory instead of create?

**Comment by nimeshgarg** [comment]
Just add a function that goes through 0 to 9 on the "TAB" press.
Add 1 to heldItem Position till 9 and then set it to 1.
Call setHeldItem on every press.

**Review by nimeshgarg — CHANGES_REQUESTED** [review]
Take a look in pull request comments

**Review by tristanduncombe — COMMENTED** [review]
Code is looking good now - only last issue is that the drag noise has a clicking at the end of it which is quite annoying - if possible fix, let me know if it isn't possible

**Review by tristanduncombe — APPROVED** [review]

**Review by tristanduncombe — APPROVED** [review]
